<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Query the GTEx Portal API • gtexr</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="apple-touch-icon-60x60.png">
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="deps/headroom-0.11.0/headroom.min.js"></script><script src="deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="deps/search-1.0.0/fuse.min.js"></script><script src="deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Query the GTEx Portal API">
<meta name="description" content="A convenient R interface to the Genotype-Tissue Expression (GTEx) Portal API. The GTEx project is a comprehensive public resource for studying tissue-specific gene expression and regulation in human tissues. Through systematic analysis of RNA sequencing data from 54 non-diseased tissue sites across nearly 1000 individuals, GTEx provides crucial insights into the relationship between genetic variation and gene expression. This data is accessible through the GTEx Portal API enabling programmatic access to human gene expression data. For more information on the API, see &lt;https://gtexportal.org/api/v2/redoc&gt;.">
<meta property="og:description" content="A convenient R interface to the Genotype-Tissue Expression (GTEx) Portal API. The GTEx project is a comprehensive public resource for studying tissue-specific gene expression and regulation in human tissues. Through systematic analysis of RNA sequencing data from 54 non-diseased tissue sites across nearly 1000 individuals, GTEx provides crucial insights into the relationship between genetic variation and gene expression. This data is accessible through the GTEx Portal API enabling programmatic access to human gene expression data. For more information on the API, see &lt;https://gtexportal.org/api/v2/redoc&gt;.">
<meta property="og:image" content="https://docs.ropensci.org/gtexr/logo.png">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="index.html">gtexr</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.2.0.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="articles/gtexr.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="articles/developer_guide.html">Developer guide</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/ropensci/gtexr/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header">
<img src="logo.png" class="logo" alt=""><h1 id="gtexr-">gtexr 
<a class="anchor" aria-label="anchor" href="#gtexr-"></a>
</h1>
</div>
<!-- badges: start -->

<!-- badges: end -->
<p>The goal of gtexr is to provide a convenient R interface to the <a href="https://gtexportal.org/api/v2/redoc#tag/GTEx-Portal-API-Info" class="external-link">GTEx Portal API V2</a>.</p>
<p>The <a href="https://www.gtexportal.org/home/" class="external-link">Genotype-Tissue Expression (GTEx) project</a> is a comprehensive public resource for studying tissue-specific gene expression and regulation in human tissues. Through systematic analysis of RNA sequencing data from 54 non-diseased tissue sites across nearly 1000 individuals, GTEx provides crucial insights into the relationship between genetic variation and gene expression. This data is accessible through the <a href="https://gtexportal.org/api/v2/redoc#tag/GTEx-Portal-API-Info" class="external-link">GTEx Portal API</a>, enabling programmatic access to human gene expression data.</p>
<p>New to R but want to explore the available data? Try out the interactive no-code ⭐<a href="https://7hocgq-rmgpanw.shinyapps.io/gtexr/" class="external-link">shiny app</a>⭐.</p>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>You can install this package from CRAN:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"gtexr"</span><span class="op">)</span></span></code></pre></div>
<p>Or you can install the development version of gtexr from either <a href="https://github.com/ropensci/gtexr" class="external-link">GitHub</a> with:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># install.packages("pak")</span></span>
<span><span class="fu">pak</span><span class="fu">::</span><span class="fu"><a href="https://pak.r-lib.org/reference/pak.html" class="external-link">pak</a></span><span class="op">(</span><span class="st">"ropensci/gtexr"</span><span class="op">)</span> <span class="co"># source - GitHub</span></span></code></pre></div>
<p>… or <a href="https://ropensci.r-universe.dev/builds" class="external-link">R Universe</a> with:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"gtexr"</span>, repos <span class="op">=</span> <span class="st">"https://ropensci.r-universe.dev"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="available-functionality">Available functionality<a class="anchor" aria-label="anchor" href="#available-functionality"></a>
</h2>
<ul>
<li>
<strong>GTEx Portal API Info</strong> – Retrieve general service information about the GTEx API.<br>
</li>
<li>
<strong>Admin Endpoints</strong> – Access maintenance messages and news updates from GTEx.<br>
</li>
<li>
<strong>Static Association Endpoints</strong> – Query precomputed eQTL and sQTL associations across tissues.<br>
</li>
<li>
<strong>Dynamic Association Endpoints</strong> – Perform on-the-fly eQTL and sQTL calculations.<br>
</li>
<li>
<strong>Biobank Data Endpoints</strong> – Retrieve metadata on biobank samples.</li>
<li>
<strong>Datasets Endpoints</strong> – Access various GTEx dataset information, as well as variant annotation and linkage disequilibrium data.</li>
<li>
<strong>Expression Data Endpoints</strong> – Obtain expression levels across tissues, including gene, exon, junction, and transcript-level data.<br>
</li>
<li>
<strong>Histology Endpoints</strong> – Retrieve tissue histology image data.<br>
</li>
<li>
<strong>Reference Genome Endpoints</strong> – Query reference genome features, including genetic coordinates for genes, transcripts and exons, as well as reported phenotype associations for a region.</li>
</ul>
</div>
<div class="section level2">
<h2 id="examples">Examples<a class="anchor" aria-label="anchor" href="#examples"></a>
</h2>
<p>Get general information about the GTEx service:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://docs.ropensci.org/gtexr/">gtexr</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="reference/get_service_info.html">get_service_info</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://pillar.r-lib.org/reference/glimpse.html" class="external-link">glimpse</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; Rows: 1</span></span>
<span><span class="co">#&gt; Columns: 9</span></span>
<span><span class="co">#&gt; $ id                &lt;chr&gt; "org.gtexportal.rest.v2"</span></span>
<span><span class="co">#&gt; $ name              &lt;chr&gt; "GTEx Portal V2 API"</span></span>
<span><span class="co">#&gt; $ version           &lt;chr&gt; "2.0.0"</span></span>
<span><span class="co">#&gt; $ organization_name &lt;chr&gt; "GTEx Project"</span></span>
<span><span class="co">#&gt; $ organization_url  &lt;chr&gt; "https://gtexportal.org"</span></span>
<span><span class="co">#&gt; $ description       &lt;chr&gt; "This service provides access to the data powering t…</span></span>
<span><span class="co">#&gt; $ contactUrl        &lt;chr&gt; "https://gtexportal.org/home/contact"</span></span>
<span><span class="co">#&gt; $ documentationUrl  &lt;chr&gt; "https://gtexportal.org/api/v2/docs"</span></span>
<span><span class="co">#&gt; $ environment       &lt;chr&gt; "prod"</span></span></code></pre></div>
<p>Retrieve eQTL genes for whole blood tissue:<a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content="&lt;p&gt;Note the warning raised if the number of items returned by a function call exceeds the requested page size. Argument &lt;code&gt;itemsPerPage&lt;/code&gt; is set to 250 by default, but may be increased to ensure that all items are retrieved in one go.&lt;/p&gt;"><sup>1</sup></a></p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/get_eqtl_genes.html">get_eqtl_genes</a></span><span class="op">(</span><span class="st">"Whole_Blood"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: ! Total number of items (12360) exceeds the selected maximum page size (250).</span></span>
<span><span class="co">#&gt; ✖ 12110 items were not retrieved.</span></span>
<span><span class="co">#&gt; ℹ To retrieve all available items, increase `itemsPerPage`, ensuring you reuse</span></span>
<span><span class="co">#&gt;   your original query parameters e.g.</span></span>
<span><span class="co">#&gt;   `get_eqtl_genes(&lt;your_existing_parameters&gt;, itemsPerPage = 100000)`</span></span>
<span><span class="co">#&gt; ℹ Alternatively, adjust global "gtexr.itemsPerPage" setting e.g.</span></span>
<span><span class="co">#&gt;   `options(list(gtexr.itemsPerPage = 100000))`</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── Paging info ─────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt; • numberOfPages = 50</span></span>
<span><span class="co">#&gt; • page = 0</span></span>
<span><span class="co">#&gt; • maxItemsPerPage = 250</span></span>
<span><span class="co">#&gt; • totalNumberOfItems = 12360</span></span>
<span><span class="co">#&gt; # A tibble: 250 × 10</span></span>
<span><span class="co">#&gt;    tissueSiteDetailId ontologyId  datasetId empiricalPValue gencodeId geneSymbol</span></span>
<span><span class="co">#&gt;    &lt;chr&gt;              &lt;chr&gt;       &lt;chr&gt;               &lt;dbl&gt; &lt;chr&gt;     &lt;chr&gt;     </span></span>
<span><span class="co">#&gt;  1 Whole_Blood        UBERON:001… gtex_v8          1.05e- 9 ENSG0000… WASH7P    </span></span>
<span><span class="co">#&gt;  2 Whole_Blood        UBERON:001… gtex_v8          1.06e-25 ENSG0000… RP11-34P1…</span></span>
<span><span class="co">#&gt;  3 Whole_Blood        UBERON:001… gtex_v8          6.31e- 2 ENSG0000… CICP27    </span></span>
<span><span class="co">#&gt;  4 Whole_Blood        UBERON:001… gtex_v8          8.71e- 9 ENSG0000… RP11-34P1…</span></span>
<span><span class="co">#&gt;  5 Whole_Blood        UBERON:001… gtex_v8          6.01e-20 ENSG0000… RP11-34P1…</span></span>
<span><span class="co">#&gt;  6 Whole_Blood        UBERON:001… gtex_v8          6.96e- 9 ENSG0000… RP11-34P1…</span></span>
<span><span class="co">#&gt;  7 Whole_Blood        UBERON:001… gtex_v8          3.10e- 4 ENSG0000… RP11-34P1…</span></span>
<span><span class="co">#&gt;  8 Whole_Blood        UBERON:001… gtex_v8          1.92e- 3 ENSG0000… ABC7-4304…</span></span>
<span><span class="co">#&gt;  9 Whole_Blood        UBERON:001… gtex_v8          1.58e- 3 ENSG0000… RP11-34P1…</span></span>
<span><span class="co">#&gt; 10 Whole_Blood        UBERON:001… gtex_v8          7.82e- 2 ENSG0000… AP006222.2</span></span>
<span><span class="co">#&gt; # ℹ 240 more rows</span></span>
<span><span class="co">#&gt; # ℹ 4 more variables: log2AllelicFoldChange &lt;dbl&gt;, pValue &lt;dbl&gt;,</span></span>
<span><span class="co">#&gt; #   pValueThreshold &lt;dbl&gt;, qValue &lt;dbl&gt;</span></span></code></pre></div>
<p>Retrieve significant eQTLs for one or more genes:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/get_significant_single_tissue_eqtls.html">get_significant_single_tissue_eqtls</a></span><span class="op">(</span>gencodeId <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>  <span class="st">"ENSG00000132693.12"</span>,</span>
<span>  <span class="st">"ENSG00000203782.5"</span></span>
<span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── Paging info ─────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt; • numberOfPages = 1</span></span>
<span><span class="co">#&gt; • page = 0</span></span>
<span><span class="co">#&gt; • maxItemsPerPage = 250</span></span>
<span><span class="co">#&gt; • totalNumberOfItems = 249</span></span>
<span><span class="co">#&gt; # A tibble: 249 × 13</span></span>
<span><span class="co">#&gt;    snpId            pos snpIdUpper variantId  geneSymbol  pValue geneSymbolUpper</span></span>
<span><span class="co">#&gt;    &lt;chr&gt;          &lt;int&gt; &lt;chr&gt;      &lt;chr&gt;      &lt;chr&gt;        &lt;dbl&gt; &lt;chr&gt;          </span></span>
<span><span class="co">#&gt;  1 rs12128960 159343657 RS12128960 chr1_1593… CRP        8.52e-5 CRP            </span></span>
<span><span class="co">#&gt;  2 rs12132451 159344052 RS12132451 chr1_1593… CRP        7.92e-5 CRP            </span></span>
<span><span class="co">#&gt;  3 rs12136402 159347493 RS12136402 chr1_1593… CRP        7.92e-5 CRP            </span></span>
<span><span class="co">#&gt;  4 rs10908709 159350390 RS10908709 chr1_1593… CRP        7.92e-5 CRP            </span></span>
<span><span class="co">#&gt;  5 rs10908710 159351189 RS10908710 chr1_1593… CRP        7.92e-5 CRP            </span></span>
<span><span class="co">#&gt;  6 rs11265178 159359256 RS11265178 chr1_1593… CRP        9.62e-5 CRP            </span></span>
<span><span class="co">#&gt;  7 rs35532309 159360755 RS35532309 chr1_1593… CRP        6.11e-5 CRP            </span></span>
<span><span class="co">#&gt;  8 rs6692378  159369451 RS6692378  chr1_1593… CRP        1.17e-6 CRP            </span></span>
<span><span class="co">#&gt;  9 rs10908714 159370563 RS10908714 chr1_1593… CRP        1.80e-5 CRP            </span></span>
<span><span class="co">#&gt; 10 rs6656924  159372915 RS6656924  chr1_1593… CRP        1.00e-6 CRP            </span></span>
<span><span class="co">#&gt; # ℹ 239 more rows</span></span>
<span><span class="co">#&gt; # ℹ 6 more variables: datasetId &lt;chr&gt;, tissueSiteDetailId &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   ontologyId &lt;chr&gt;, chromosome &lt;chr&gt;, gencodeId &lt;chr&gt;, nes &lt;dbl&gt;</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="citing-gtexr">Citing gtexr<a class="anchor" aria-label="anchor" href="#citing-gtexr"></a>
</h2>
<p>If you find gtexr useful, please consider citing both GTEx and gtexr. Citation details are available <a href="https://docs.ropensci.org/gtexr/authors.html#citation">here</a>.</p>
<p>Example publications citing gtexr include:</p>
<ul>
<li><a href="https://pubmed.ncbi.nlm.nih.gov/40064387/" class="external-link">Schwartz R, Warwick AN, et al. Genetic Distinctions Between Reticular Pseudodrusen and Drusen: A Genome-Wide Association Study. Am J Ophthalmol. 2025 Mar 8:S0002-9394(25)00119-9. doi: 10.1016/j.ajo.2025.03.007. Epub ahead of print. PMID: 40064387</a></li>
</ul>
</div>
<div class="section level2">
<h2 id="community-guidelines">Community guidelines<a class="anchor" aria-label="anchor" href="#community-guidelines"></a>
</h2>
<p>Feedback, bug reports, and feature requests are welcome; file issues or seek support <a href="https://github.com/ropensci/gtexr/issues" class="external-link">here</a>. If you would like to contribute to the package, please see our <a href="https://docs.ropensci.org/gtexr/CONTRIBUTING.html">contributing guidelines</a>.</p>
<p>Please note that this package is released with a <a href="https://ropensci.org/code-of-conduct/" class="external-link">Contributor Code of Conduct</a>. By contributing to this project, you agree to abide by its terms.</p>
</div>
</div>


  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://cloud.r-project.org/package=gtexr" class="external-link">View on CRAN</a></li>
<li><a href="https://github.com/ropensci/gtexr/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/ropensci/gtexr/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>

<div class="community">
<h2 data-toc-skip>Community</h2>
<ul class="list-unstyled">
<li><a href="CONTRIBUTING.html">Contributing guide</a></li>
</ul>
</div>

<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing gtexr</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Alasdair Warwick <br><small class="roles"> Author, maintainer, copyright holder </small> <a href="https://orcid.org/0000-0002-0800-2890" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Benjamin Zuckerman <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0002-0077-6074" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Abraham Olvera-Barrios <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0002-3305-4465" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Chuin Ying Ung <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0001-8487-4589" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Robert Luben <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0002-5088-6343" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li><a href="authors.html">More about authors...</a></li>
</ul>
</div>

<div class="dev-status">
<h2 data-toc-skip>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/ropensci/gtexr/actions/workflows/pkgdown.yaml" class="external-link"><img src="https://github.com/ropensci/gtexr/actions/workflows/pkgdown.yaml/badge.svg" alt="pkgdown"></a></li>
<li><a href="https://app.codecov.io/gh/ropensci/gtexr?branch=main" class="external-link"><img src="https://codecov.io/gh/ropensci/gtexr/branch/main/graph/badge.svg" alt="Codecov test coverage"></a></li>
<li><a href="https://github.com/ropensci/gtexr/actions/workflows/R-CMD-check.yaml" class="external-link"><img src="https://github.com/ropensci/gtexr/actions/workflows/R-CMD-check.yaml/badge.svg" alt="R-CMD-check"></a></li>
<li><a href="https://github.com/ropensci/gtexr/actions/workflows/shinyapps-deploy.yaml" class="external-link"><img src="https://github.com/ropensci/gtexr/actions/workflows/shinyapps-deploy.yaml/badge.svg" alt="Deploy to shinyapps.io"></a></li>
<li><a href="https://CRAN.R-project.org/package=gtexr" class="external-link"><img src="https://www.r-pkg.org/badges/version/gtexr" alt="CRAN status"></a></li>
<li><a href="https://CRAN.R-project.org/package=gtexr" class="external-link"><img src="https://cranlogs.r-pkg.org/badges/gtexr" alt="CRAN downloads"></a></li>
<li><a href="https://www.repostatus.org/#active" class="external-link"><img src="https://www.repostatus.org/badges/latest/active.svg" alt="Repo Status"></a></li>
<li><a href="https://github.com/ropensci/software-review/issues/684" class="external-link"><img src="https://badges.ropensci.org/684_status.svg" alt="Status at rOpenSci Software Peer Review"></a></li>
</ul>
</div>

  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Alasdair Warwick, Benjamin Zuckerman, Abraham Olvera-Barrios, Chuin Ying Ung, Robert Luben.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
