<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Get Multi Tissue Eqtls — get_multi_tissue_eqtls • gtexr</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Get Multi Tissue Eqtls — get_multi_tissue_eqtls"><meta name="description" content="Find multi-tissue eQTL Metasoft results.
This service returns multi-tissue eQTL Metasoft results for a given gene and variant in a specified dataset.
A Versioned GENCODE ID must be provided.
For each tissue, the results include: m-value (mValue), normalized effect size (nes), p-value (pValue), and standard error (se).
The m-value is the posterior probability that an eQTL effect exists in each tissue tested in the cross-tissue meta-analysis (Han and Eskin, PLoS Genetics 8(3): e1002555, 2012).
The normalized effect size is the slope of the linear regression of normalized expression data versus the three genotype categories using single-tissue eQTL analysis, representing eQTL effect size.
The p-value is from a t-test that compares observed NES from single-tissue eQTL analysis to a null NES of 0.


By default, the service queries the latest GTEx release. The retrieved data is split into pages with items_per_page entries per page
GTEx Portal API documentation"><meta property="og:description" content="Find multi-tissue eQTL Metasoft results.
This service returns multi-tissue eQTL Metasoft results for a given gene and variant in a specified dataset.
A Versioned GENCODE ID must be provided.
For each tissue, the results include: m-value (mValue), normalized effect size (nes), p-value (pValue), and standard error (se).
The m-value is the posterior probability that an eQTL effect exists in each tissue tested in the cross-tissue meta-analysis (Han and Eskin, PLoS Genetics 8(3): e1002555, 2012).
The normalized effect size is the slope of the linear regression of normalized expression data versus the three genotype categories using single-tissue eQTL analysis, representing eQTL effect size.
The p-value is from a t-test that compares observed NES from single-tissue eQTL analysis to a null NES of 0.


By default, the service queries the latest GTEx release. The retrieved data is split into pages with items_per_page entries per page
GTEx Portal API documentation"><meta property="og:image" content="https://docs.ropensci.org/gtexr/logo.png"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">gtexr</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.2.0.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/gtexr.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/developer_guide.html">Developer guide</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/ropensci/gtexr/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Get Multi Tissue Eqtls</h1>
      <small class="dont-index">Source: <a href="https://github.com/ropensci/gtexr/blob/main/R/get_multi_tissue_eqtls.R" class="external-link"><code>R/get_multi_tissue_eqtls.R</code></a></small>
      <div class="d-none name"><code>get_multi_tissue_eqtls.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Find multi-tissue eQTL <code>Metasoft</code> results.</p><ul><li><p>This service returns multi-tissue eQTL Metasoft results for a given gene and variant in a specified dataset.</p></li>
<li><p>A Versioned GENCODE ID must be provided.</p></li>
<li><p>For each tissue, the results include: m-value (mValue), normalized effect size (nes), p-value (pValue), and standard error (se).</p></li>
<li><p>The m-value is the posterior probability that an eQTL effect exists in each tissue tested in the cross-tissue meta-analysis (Han and Eskin, PLoS Genetics 8(3): e1002555, 2012).</p></li>
<li><p>The normalized effect size is the slope of the linear regression of normalized expression data versus the three genotype categories using single-tissue eQTL analysis, representing eQTL effect size.</p></li>
<li><p>The p-value is from a t-test that compares observed NES from single-tissue eQTL analysis to a null NES of 0.</p></li>
</ul><p>By default, the service queries the latest GTEx release. The retrieved data is split into pages with <code>items_per_page</code> entries per page</p>
<p><a href="https://gtexportal.org/api/v2/redoc#tag/Static-Association-Endpoints/operation/get_multi_tissue_eqtls_api_v2_association_metasoft_get" class="external-link">GTEx Portal API documentation</a></p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">get_multi_tissue_eqtls</span><span class="op">(</span></span>
<span>  <span class="va">gencodeId</span>,</span>
<span>  variantId <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  datasetId <span class="op">=</span> <span class="st">"gtex_v8"</span>,</span>
<span>  page <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  itemsPerPage <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"gtexr.itemsPerPage"</span><span class="op">)</span>,</span>
<span>  .verbose <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"gtexr.verbose"</span><span class="op">)</span>,</span>
<span>  .return_raw <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-gencodeid">gencodeId<a class="anchor" aria-label="anchor" href="#arg-gencodeid"></a></dt>
<dd><p>String. A Versioned GENCODE ID of a gene, e.g.
"ENSG00000065613.9".</p></dd>


<dt id="arg-variantid">variantId<a class="anchor" aria-label="anchor" href="#arg-variantid"></a></dt>
<dd><p>String. A gtex variant ID.</p></dd>


<dt id="arg-datasetid">datasetId<a class="anchor" aria-label="anchor" href="#arg-datasetid"></a></dt>
<dd><p>String. Unique identifier of a dataset. Usually includes a
data source and data release. Options: "gtex_v8", "gtex_snrnaseq_pilot".</p></dd>


<dt id="arg-page">page<a class="anchor" aria-label="anchor" href="#arg-page"></a></dt>
<dd><p>Integer (default = 0).</p></dd>


<dt id="arg-itemsperpage">itemsPerPage<a class="anchor" aria-label="anchor" href="#arg-itemsperpage"></a></dt>
<dd><p>Integer (default = 250). Set globally to maximum value
100000 with <code>options(list(gtexr.itemsPerPage = 100000))</code>.</p></dd>


<dt id="arg--verbose">.verbose<a class="anchor" aria-label="anchor" href="#arg--verbose"></a></dt>
<dd><p>Logical. If <code>TRUE</code> (default), print paging information. Set
to <code>FALSE</code> globally with <code>options(list(gtexr.verbose = FALSE))</code>.</p></dd>


<dt id="arg--return-raw">.return_raw<a class="anchor" aria-label="anchor" href="#arg--return-raw"></a></dt>
<dd><p>Logical. If <code>TRUE</code>, return the raw API JSON response.
Default = <code>FALSE</code></p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A tibble. Or a list if <code>.return_raw = TRUE</code>.</p>
    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p>Other Static Association Endpoints:
<code><a href="get_eqtl_genes.html">get_eqtl_genes</a>()</code>,
<code><a href="get_fine_mapping.html">get_fine_mapping</a>()</code>,
<code><a href="get_independent_eqtl.html">get_independent_eqtl</a>()</code>,
<code><a href="get_significant_single_tissue_eqtls.html">get_significant_single_tissue_eqtls</a>()</code>,
<code><a href="get_significant_single_tissue_eqtls_by_location.html">get_significant_single_tissue_eqtls_by_location</a>()</code>,
<code><a href="get_significant_single_tissue_ieqtls.html">get_significant_single_tissue_ieqtls</a>()</code>,
<code><a href="get_significant_single_tissue_isqtls.html">get_significant_single_tissue_isqtls</a>()</code>,
<code><a href="get_significant_single_tissue_sqtls.html">get_significant_single_tissue_sqtls</a>()</code>,
<code><a href="get_sqtl_genes.html">get_sqtl_genes</a>()</code></p></div>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># \dontrun{</span></span></span>
<span class="r-in"><span><span class="co"># search by gene</span></span></span>
<span class="r-in"><span><span class="fu">get_multi_tissue_eqtls</span><span class="op">(</span>gencodeId <span class="op">=</span> <span class="st">"ENSG00000132693.12"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Paging info</span> <span style="color: #00BBBB;">─────────────────────────────────────────────────────────────────</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> numberOfPages = 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> page = 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> maxItemsPerPage = 250</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> totalNumberOfItems = 93</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 93 × 5</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    gencodeId          datasetId    metaP variantId               tissues </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>              <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                   <span style="color: #949494; font-style: italic;">&lt;list&gt;</span>  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 1</span> ENSG00000132693.12 gtex_v8   0.001<span style="text-decoration: underline;">42</span>  chr1_159476920_T_C_b38  <span style="color: #949494;">&lt;tibble&gt;</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 2</span> ENSG00000132693.12 gtex_v8   0.092<span style="text-decoration: underline;">7</span>   chr1_159245536_C_T_b38  <span style="color: #949494;">&lt;tibble&gt;</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 3</span> ENSG00000132693.12 gtex_v8   0.001<span style="text-decoration: underline;">58</span>  chr1_159439013_C_T_b38  <span style="color: #949494;">&lt;tibble&gt;</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 4</span> ENSG00000132693.12 gtex_v8   0.000<span style="text-decoration: underline;">735</span> chr1_159441913_T_A_b38  <span style="color: #949494;">&lt;tibble&gt;</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 5</span> ENSG00000132693.12 gtex_v8   0.001<span style="text-decoration: underline;">58</span>  chr1_159471651_G_A_b38  <span style="color: #949494;">&lt;tibble&gt;</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 6</span> ENSG00000132693.12 gtex_v8   0.001<span style="text-decoration: underline;">58</span>  chr1_159396484_C_T_b38  <span style="color: #949494;">&lt;tibble&gt;</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 7</span> ENSG00000132693.12 gtex_v8   0.001<span style="text-decoration: underline;">58</span>  chr1_159386699_AT_A_b38 <span style="color: #949494;">&lt;tibble&gt;</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 8</span> ENSG00000132693.12 gtex_v8   0.002<span style="text-decoration: underline;">63</span>  chr1_159545608_A_T_b38  <span style="color: #949494;">&lt;tibble&gt;</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 9</span> ENSG00000132693.12 gtex_v8   0.001<span style="text-decoration: underline;">42</span>  chr1_159476580_G_A_b38  <span style="color: #949494;">&lt;tibble&gt;</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">10</span> ENSG00000132693.12 gtex_v8   0.001<span style="text-decoration: underline;">66</span>  chr1_159505857_T_A_b38  <span style="color: #949494;">&lt;tibble&gt;</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ 83 more rows</span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># note that 'tissues' is a list-column</span></span></span>
<span class="r-in"><span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu">get_multi_tissue_eqtls</span><span class="op">(</span>gencodeId <span class="op">=</span> <span class="st">"ENSG00000132693.12"</span>,</span></span>
<span class="r-in"><span>                            variantId <span class="op">=</span> <span class="st">"chr1_159476920_T_C_b38"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Paging info</span> <span style="color: #00BBBB;">─────────────────────────────────────────────────────────────────</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> numberOfPages = 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> page = 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> maxItemsPerPage = 250</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> totalNumberOfItems = 1</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">x</span><span class="op">$</span><span class="va">tissues</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 49 × 5</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    tissue                          mValue    pValue     se     nes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                            <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 1</span> Thyroid                          0.983 0.000<span style="text-decoration: underline;">029</span>2 0.059<span style="text-decoration: underline;">3</span>  0.250 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 2</span> Testis                           0.342 0.040<span style="text-decoration: underline;">4</span>    0.083<span style="text-decoration: underline;">5</span> -<span style="color: #BB0000;">0.172</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 3</span> Small_Intestine_Terminal_Ileum   0.551 0.864     0.122  -<span style="color: #BB0000;">0.021</span><span style="color: #BB0000; text-decoration: underline;">1</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 4</span> Brain_Frontal_Cortex_BA9         0.489 0.204     0.113  -<span style="color: #BB0000;">0.144</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 5</span> Skin_Not_Sun_Exposed_Suprapubic  0.552 0.806     0.065<span style="text-decoration: underline;">4</span>  0.016<span style="text-decoration: underline;">0</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 6</span> Vagina                           0.514 0.437     0.130  -<span style="color: #BB0000;">0.102</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 7</span> Whole_Blood                      0.494 0.632     0.040<span style="text-decoration: underline;">7</span>  0.019<span style="text-decoration: underline;">5</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 8</span> Breast_Mammary_Tissue            0.679 0.395     0.074<span style="text-decoration: underline;">6</span>  0.063<span style="text-decoration: underline;">6</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 9</span> Pituitary                        0.421 0.175     0.097<span style="text-decoration: underline;">2</span> -<span style="color: #BB0000;">0.132</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">10</span> Minor_Salivary_Gland             0.678 0.349     0.126   0.118 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ 39 more rows</span></span>
<span class="r-in"><span><span class="co"># }</span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Alasdair Warwick, Benjamin Zuckerman, Abraham Olvera-Barrios, Chuin Ying Ung, Robert Luben.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

